sip-plus-perinatal/index.ts

```diff
@@ -15,25 +16,32 @@ router.group('/perinatal', (group) => {
     group.post('/register', async (req: any, res) => {
         res.send({ message: 'ok' });
         try {
-            const prestacion = req.body.data;
-            // verificamos si la prestacion es "Consulta de control de embarazo"
-            // "consulta de ginecología", "Consulta de obstetricia", "Consulta de tocoginecología" o "consulta de control de embarazo de alto riesgo"
-            const conceptConsultaEmbarazo = ['1191000013107', '721000013105', '1181000013109', '1141000013103',
+            const prestacion = req.body.data.prestacion;
+            // Se asume que el registro de número de "embarazo" está incluido en todos los conceptConsultaEmbarazo.
+            const conceptConsultaEmbarazo = ['1191000013107', '1141000013103',
                 '1341000013100', '1201000013105', '1741000013104', '4501000013105', '1711000013103'];
             const conceptId = prestacion.solicitud.tipoPrestacion.conceptId;
-
-            if (conceptConsultaEmbarazo.includes(conceptId)) {
+            // verificamos si la prestación es algunas de las relacionadas a control de embarazo
+            // si no lo es, entonces verificamos que contenga entre sus registros alguno de estos conceptos
+            let mapeoSipPlus: boolean = conceptConsultaEmbarazo.includes(conceptId);
+            const registros = getRegistros(prestacion.ejecucion.registros);
+            if (!mapeoSipPlus) {
+                // Se verifica que la prestación incluya algún registro relacionado a control de embarazo
+                mapeoSipPlus = registros.find(reg => conceptConsultaEmbarazo.includes(reg.concepto.conceptId));
+            }
+            if (mapeoSipPlus) {
                 let paciente: any = await getPacienteAndes(prestacion.paciente.id);
                 if (paciente) {
                     // Obtenemos el paciente y sus gestas cargadas en sip plus
                     const resultSP = await getPacienteSP(paciente);
                     if (resultSP) {
                         // peticion válida
-                        await postPacienteSP(paciente, prestacion, resultSP.paciente || {});
+                        await postPacienteSP(paciente, prestacion, registros, resultSP.paciente || {});
                     }
                 }
             }
         } catch (e) {
+            log(fakeRequest, 'microservices:integration:sip-plus', { data: req.body.data }, 'postInitial:register', e);
             throw e;
         }
```

Comment:
Hay que remover `throw e;`. También fijate que no están bien los parámetros para el log. (`(alias) log(req: IRequest, key: String, paciente: any, operacion: String, valor: any, anterior?: any, error?: any): Promise<any>`)
```suggestion
        } catch (e) {
            log(fakeRequest, 'microservices:integration:sip-plus', { data: req.body.data }, 'postInitial:register', e);
        }
```